File: app/src/main/java/com/example/roombooking/fragments/SplashScreenFragment.java
package com.example.roombooking.fragments;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.roombooking.R;
import com.example.roombooking.activities.MainActivity;
import com.example.roombooking.manager.SharedPreferencesManager;

public class SplashScreenFragment extends BaseFragment
{
	@Override
	protected int getLayoutContentID()
	{
		return R.layout.fragment_splash_screen;
	}

	@Override
	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState)
	{
		super.onViewCreated(view, savedInstanceState);

		Runnable runnable = new Runnable()
		{
			@Override
			public void run()
			{
				String token = SharedPreferencesManager.getInstance(getActivity()).getUserToken();
				if(SharedPreferencesManager.getInstance(getActivity()).getUserToken() != null)
				{
					startActivity(new Intent(getActivity(), MainActivity.class));
					getActivity().finish();				}
				else
				{
					switchFragment(new LoginScreenFragment(), null, false);
				}
			}
		};

		Handler checkUserTokenHandler = new Handler();
		checkUserTokenHandler.postDelayed(runnable, 2000);
	}
}

File: app/src/main/java/com/example/roombooking/fragments/SplashScreenFragment.kt
package com.example.roombooking.fragments

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.View
import com.example.roombooking.R
import com.example.roombooking.activities.MainActivity
import com.example.roombooking.manager.SharedPreferencesManager

class SplashScreenFragment : BaseFragment() {
    override val layoutContentID: Int = R.layout.fragment_splash_screen

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val runnable = Runnable {
           activity?.let {
               if (SharedPreferencesManager.getInstance(it)!!.userToken != null) {
                   startActivity(Intent(activity, MainActivity::class.java))
                   activity!!.finish()
               } else {
                   switchFragment<BaseFragment>(LoginScreenFragment(), null, false)
               }
           }
        }
        val checkUserTokenHandler = Handler()
        checkUserTokenHandler.postDelayed(runnable, 2000)
    }
}